src/tintx/objects.py

```diff
@@ -262,14 +261,22 @@ def get_object_prop(
                 ygrid_diff,
             )
             fwd = transform.Affine.from_gdal(*geotransform)
-            # use safe dtype in case of num_cells > 255
+
+            # iterate over shapes (we have only two) until
+            # we have the correct one (with num == 1)
+            # see https://github.com/antarcticrainforest/tintX/issues/79
+            def get_shape(shapes: Iterator[tuple]) -> Any:
+                shp, sid = next(shapes)
+                while not sid:
+                    shp, sid = next(shapes)
+                return geometry.shape(shp)
```

Comment:
- Putting an all-0 array would be problematic, right? As it would try to iterate over an empty iterator. How abut:

```suggestion
                for shp, sid in shapes:
                    if sid:
                       return geometry.shape(shp)
```

If there is no shape then I assume you also don't want to add it to the polygons list?

- Could you also annotate the types that the tuple holds?  I assume something like: `Tuple[Dict[str, Union[str, List[List[Tuple[float,...]]]]], float]` Although that looks a little ridiculous, maybe there are some better types.

- Also, the return type shouldn't be really `Any` but `geometry.polygon.Polygon`

Sorry I have become kind of annoying when it comes to type annotations.

Reply:
> Sorry I have become kind of annoying when it comes to type annotations.

No worries, I'm trying to adapt. But shapely doesn't have typing yet, I used the one which reveal_type was suggesting. I'd really appreciate if you could fill in the voids here until I'm a bit more versed on that topic.